test/playlist.test.js

```diff
@@ -46,17 +46,27 @@ describe('Playlist', () => {
 			sinon.assert.calledOnce(player.update);
 		});
 
-		it('listens for the video to end to trigger the next in the queue', () => {
+		it('listens for the video to end to trigger the next in the queue when autoplay is set', () => {
 			player.videoData = { id: 'bar' };
 
-			new Subject({ player, queue });
+			new Subject({ player, queue, autoplay: true });
 
 			// no DOM so trigger this on the listener directly
 			player.containerEl.dispatchEvent(new CustomEvent('ended', { bubbles: false }));
 
 			sinon.assert.calledOnce(player.update);
 			sinon.assert.calledWith(player.update, sinon.match({ id: 'baz' }));
 		});
+
+		it('doesn\'t listen for the video to end when autoplay is not set', () => {
+			player.videoData = { id: 'bar' };
+
+			new Subject({ player, queue, autoplay: false });
+
+			player.containerEl.dispatchEvent(new CustomEvent('ended', { bubbles: false }));
+
+			sinon.assert.notCalled(player.update);
```

Comment:
I think this will always pass, even if autoplay was set to `true` because `this.update` isn't the function being called in the eventhandler, it is actually `this.next.bind(this)` which returns a new function.